--- orginal/pattern.cpp	2009-04-14 00:16:33.000000000 +0700
+++ pattern.cpp	2009-06-12 10:42:40.000000000 +0700
@@ -19,7 +19,7 @@
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 --------------------------------------------------------------------------------*/
-#include "stdafx.h"
+
 #include "pattern.h"
 
 //////////////////////////////////////////////////
